### PR TITLE
add s3 processor

### PR DESCRIPTION
**Pull request by jem-davies**
Couple of slack / discord questions about S3 SQS this PR should help with: 

This `aws_s3` processor is offered as an alternative to [streaming-objects-on-upload-with-sqs](https://warpstreamlabs.github.io/bento/docs/components/inputs/aws_s3#streaming-objects-on-upload-with-sqs).
This `aws_s3` processor may be preferable to the `aws_s3` input with the field `sqs` in the following situations:

- You require data from the SQS message as well the S3 Object Data
- You are using a similar pattern to the `streaming-objects-on-upload-with-sqs` but with a different queue technology such as RabbitMQ
- You need to access some data from S3 as part of the processing stage of your config

### CODE REVIEW

**Comment by jem-davies** [comment]
https://github.com/warpstreamlabs/bento/issues/173

File: internal/impl/aws/processor_s3.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	})
+
+	var resBatches []service.MessageBatch
+
+	for i := range batch {
+
+		bucket, err := bucketExecutor.TryString(i)
+		if err != nil {
+			return nil, fmt.Errorf("s3 bucket interpolation error: %w", err)
+		}
```
[the remark this answers is not in the document]
Comment (jem-davies) [reply internal/impl/aws/processor_s3.go:179]: yep brain fart here

File: internal/impl/aws/processor_s3_test.go
[the remark this answers is not in the document]
Comment (jem-davies) [reply internal/impl/aws/processor_s3_test.go:1]: updated

File: internal/impl/aws/processor_s3.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+		if obj.LastModified != nil {
+			part.MetaSetMut("s3_last_modified", obj.LastModified.Format(time.RFC3339))
+			part.MetaSetMut("s3_last_modified_unix", obj.LastModified.Unix())
+		}
+		if obj.ContentType != nil {
+			part.MetaSetMut("s3_content_type", *obj.ContentType)
+		}
+		if obj.ContentEncoding != nil {
+			part.MetaSetMut("s3_content_encoding", *obj.ContentEncoding)
+		}
```
[the remark this answers is not in the document]
Comment (jem-davies) [reply internal/impl/aws/processor_s3.go:240]: added in

File: internal/impl/aws/processor_s3.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jem-davies) [reply internal/impl/aws/processor_s3.go]: updated the error handling

**Review by gregfurman — APPROVED** [review]
LGTM!

**Pull request merged**